src/wallet/psbtwallet.cpp

```diff
@@ -24,8 +24,7 @@ bool FillPSBT(const CWallet* pwallet, PartiallySignedTransaction& psbtx, Transac
 
         // Get the Sighash type
         if (sign && input.sighash_type > 0 && input.sighash_type != sighash_type) {
-            error = TransactionError::SIGHASH_MISMATCH;
-            return false;
+            return TransactionError::SIGHASH_MISMATCH;
```

Comment:
It is present in the original PR bitcoin#14978 : https://github.com/bitcoin/bitcoin/pull/14978/files#diff-ae9c5829347ce7ec51af0fff0a2e73f62dbe338363a79474eb0c6e014d575fa7R21-R24

But that piece is missing from our backport in #4229: https://github.com/dashpay/dash/pull/4229/files#diff-ae9c5829347ce7ec51af0fff0a2e73f62dbe338363a79474eb0c6e014d575fa7R12-R23

Reading through the comments in #4229, I don't see anything related to `psbtwallet.cpp`. 

I guess @kittywhiskers would be the best person to help here. Do you happen to remember why these lines have been excluded from the backport?

Reply:
@dzutte-cpp 

The reason those PRs don't match _as-is_ is because the backport attempted to reproduce the refactor and since we didn't backport certain changes at the time of the refactor (for example, https://github.com/bitcoin/bitcoin/pull/14588, which refactors signing logic and introduces `PSBTInputSigned`) was not backported, backporting this PR (among possible others) would bridge the gap.

Reply:
Thanks @kittywhiskers , that makes sense.
I'll try to backport bitcoin#14588 and see if any others are necessary, hope that helps.

Reply:
@PastaPastaPasta @kittywhiskers I found that the missing block with `if (!input.IsSane()) {` has likely been deliberately omitted during backporting.
Here is how `PSBTInput::IsSane()` looks: https://github.com/bitcoin/bitcoin/blob/0.18/src/psbt.cpp#L157-L167
All checks in it have to do with witness scripts. Dash doesn't support witnesses, so this makes no sense for us.

I looked into backporting bitcoin#14588 but that's not trivial as its changes has further been reshuffled in bitcoin#14978 which is (partially?) merged to Dash. Let's tackle it later separately, it should not be a blocker here.

Reply:
Yeah, the changes I made were mostly related to removing SegWit stuff but I think documenting them in the PR itself would've saved everyone some effort in reading through the code manually.